client/my-sites/email/email-forwarding/email-forwarding-add-new-compact.jsx

```diff
@@ -0,0 +1,188 @@
+import { Button, Gridicon } from '@automattic/components';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import FormButton from 'calypso/components/forms/form-button';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormInputValidation from 'calypso/components/forms/form-input-validation';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+import { validateAllFields } from 'calypso/lib/domains/email-forwarding';
+import formState from 'calypso/lib/form-state';
+
+class EmailForwardingAddNewCompact extends Component {
+	static propTypes = {
+		fields: PropTypes.object,
+		index: PropTypes.number,
+		onRemoveEmailForward: PropTypes.func.isRequired,
+		selectedDomainName: PropTypes.string.isRequired,
+		onUpdateEmailForward: PropTypes.func.isRequired,
+		emailForwards: PropTypes.array,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			formSubmitting: false,
+			fields: this.props.fields,
+		};
+
+		this.formStateController = formState.Controller( {
+			initialFields: this.getInitialFields(),
+			onNewState: this.setFormState,
+			validatorFunction: ( fieldValues, onComplete ) => {
+				onComplete( null, validateAllFields( fieldValues, this.props.emailForwards ?? [] ) );
+			},
+		} );
+	}
+
+	getInitialFields() {
+		return this.props.fields;
+	}
+
+	setFormState = ( fields ) => {
+		this.setState( { fields } );
+	};
+
+	addButton() {
+		const { onButtonClick, translate } = this.props;
+		return (
+			<div className="email-forwarding-add-new-compact__actions">
+				<Button
+					className="email-forwarding-add-new-compact__add-another-forward-button"
+					onClick={ this.onForwardAdd }
```

Comment:
There is no event handler named `onForwardAdd()` in this component.

Reply:
I've updated this code to call a function supplied via a prop.